azkaban-common/src/main/java/azkaban/executor/ExecutionOptions.java

```diff
@@ -65,7 +66,7 @@
   private Integer pipelineExecId = null;
   private Integer queueLevel = 0;
   private String concurrentOption = CONCURRENT_OPTION_IGNORE;
-  private String mailCreator = DefaultMailCreator.DEFAULT_MAIL_CREATOR;
+  private String mailCreator = TemplateBasedMailCreator.NAME;
```

Comment:
I don't think it's a good idea to change default value of mailCreator to "template-based". If templateBasedMailCreator is not registered, then when you get mailCreator, you are actually getting the defaultMailCreator, but the name would be "template-based", right?

Reply:
correct; will change it to obtain most recent creator from the `MailCreatorRegistry`, ok?

Reply:
From my observation, if you register template-based mailCreator first and it is set in flow object and stored in DB. Next time when you restart the server and load the flow objects from DB, even if you don't register template_based mailCreator, the String mailCreator will still be "template-based".
I would prefer simpler approach which is to keep mailCreator as "default". Only set it to template-based when user explicitly sets it in executionOptions from web UI.

Reply:
hi @jamiesjc ! If I understand you correctly, we discuss the case when a user e.g. schedules a regular execution of a flow and the executor server somehow afterwards loses the emailCreator, is that correct? It seems UI option does not help in that case either (user can chose from a list of creators available at time T1, but the server alters its configuration at time T2>T1). What are our options?
From the other side: is there a good reason for having emailCreator in an `ExecutionOption`? Maybe it should be decided on the level of the executor server once for all executions during the initialization step (and at the time of load of the email creators). Was it considered?
My intention was not to let executing user chose the layouts ad hoc, but rather to provide the tool to evolve the content of emails outside the code-base.

Reply:
@mtrna Your proposal sounds reasonable to me. 
Now we don't have the UI support for users to specify emailCreator like other execution options. We could provide them with the flexibility to choose from a list of emailCreators in the future but I don't think it's quite necessary for now.
You are right the execution options only apply to the following execution once, and next time when you submit the flow again it will disappear. But if you schedule the flow instead, the execution options will persist with every scheduled execution.  
I'm in favor of the logic to decide the emailCreator on the level of executor during initialization step. If email templates are configured, then we use template-based emailCreator, otherwise we use the default emailCreator. 
@HappyRay Does it sound like a good approach?